src/archBuild_sconscript

```diff
@@ -38,31 +38,52 @@ sources = [
 	"updateCheck.cpp",
 ]
 
+useMsVC = env["PLATFORM"] == "win32" and env["TARGET_ARCH"] == "arm64"
 if env["PLATFORM"] == "win32":
-	# On Windows, OSARA is build with LLVM to have a toolchain that's closer to what's used on Mac
-	env["CC"] = "clang-cl"
-	env["LINK"] = "lld-link"
+	# On non ARM64EC Windows, OSARA is build with LLVM to have a toolchain that's closer to what's used on Mac
+	if not useMsVC:
+		env["CC"] = "clang-cl"
+		env["LINK"] = "lld-link"
 	env.Append(CCFLAGS=[
 		"/W3",
 		"/WX",
-		# Clang warns about unused functions in WDL
-		"/clang:-Wno-unused-function",
 	])
 	env.Append(CXXFLAGS=[
-		"/clang:-std=c++20",
+		"/std:c++20",
 		"/EHsc",
-		# Ignore suggestions to add braces around initialization of subobjects. We might want to fix this at some point
-		"/clang:-Wno-missing-braces",
-	])
-	env.Append(CFLAGS=[
-		# Ignore warnings in WDL/win32_utf8.c, not our code
-		"/clang:-Wno-deprecated-declarations",
-		"/clang:-Wno-unused-variable"
 	])
+	if not useMsVC:
+		env.Append(CCFLAGS=[
+			# Clang warns about unused functions in WDL
+			"/clang:-Wno-unused-function",
+		])
+		env.Append(CXXFLAGS=[
+			# Ignore suggestions to add braces around initialization of subobjects. We might want to fix this at some point
+			"/clang:-Wno-missing-braces",
+		])
+		env.Append(CFLAGS=[
+			# Ignore warnings in WDL/win32_utf8.c, not our code
+			"/clang:-Wno-deprecated-declarations",
+			"/clang:-Wno-unused-variable"
+		])
+	else:
+		env.Append(CPPDEFINES=[
+			"_CRT_SECURE_NO_DEPRECATE",
+			"_CRT_NONSTDC_NO_DEPRECATE",
+		])
+		env.Append(CXXFLAGS=[
+			# #479: fmt/format.h includes UTF-8 characters in comments. However, cl
+			# treats the file as ANSI unless it has a BOM. This causes a warning (and
+			# thus an error) on systems with certain locales. Just disable this warning.
+			"/wd4819",
```

Comment:
I'm trying to move towards disabling warnings for includes using compiler directives at the location of the include where reasonable. Where a file is included more than once, this doesn't make a lot of sense, but I think we only include fmt/format.h from translation.h. That said, there are plenty of other places where we just do it in sconscript, so this is totally fine if you don't want to/can't deal with that here.

Reply:
It is a bit difficult to do with this for both MSVC and Clang. May be we should just wait for clang 21 to be release with Visual Studio. I think that will be able to create ARM64EC.

Reply:
We can live with this for now. If clang 21 does bring the needed goodness, we can drop msvc then.

That said, we need to find someone with an ARM64 system to test this, so we may be blocked on that for a bit.